src/DIRAC/Core/Tornado/Server/HandlerManager.py

```diff
@@ -50,109 +52,221 @@ class HandlerManager(object):
+                                                                              handlerInstance, instName))
+                if isHTTPS and isHTTPS.lower() == 'https':
+                  urls.append(newInst)
+              else:
+                urls.append(newInst)
         # On systems sometime you have things not related to services...
         except RuntimeError:
           pass
-    return self.loadHandlersByServiceName(serviceList)
+    return urls
 
-  def loadHandlersByServiceName(self, servicesNames):
+  def loadServicesHandlers(self, services=None):
     """
       Load a list of handler from list of service using DIRAC moduleLoader
       Use :py:class:`DIRAC.Core.Base.private.ModuleLoader`
 
-      :param servicesNames: list of service, e.g. ['Framework/Hello', 'Configuration/Server']
+      :param services: List of service handlers to load. Default value set at initialization
+          If ``True``, loads all services from CS
+      :type services: bool or list
+
+      :return: S_OK()/S_ERROR()
+    """
+    # list of services, e.g. ['Framework/Hello', 'Configuration/Server']
+    if isinstance(services, string_types):
+      services = [services]
+    # list of services
+    self.__services = self.__services if services is None else services if services else []
+
+    if self.__services is True:
+      self.__services = self.discoverHandlers('Services')
+
+    if self.__services:
+      # Extract ports
+      ports, self.__services = self.__extractPorts(self.__services)
+
+      self.loader = ModuleLoader("Service", PathFinder.getServiceSection, RequestHandler, moduleSuffix="Handler")
+
+      # Use DIRAC system to load: search in CS if path is given and if not defined
+      # it search in place it should be (e.g. in DIRAC/FrameworkSystem/Service)
+      load = self.loader.loadModules(self.__services)
+      if not load['OK']:
+        return load
+      for module in self.loader.getModules().values():
+        url = module['loadName']
+
+        # URL can be like https://domain:port/service/name or just service/name
+        # Here we just want the service name, for tornado
+        serviceTuple = url.replace('https://', '').split('/')[-2:]
+        url = "%s/%s" % (serviceTuple[0], serviceTuple[1])
+        self.__addHandler(module['loadName'], module['classObj'], url, ports.get(module['modName']))
+    return S_OK()
+
+  def __extractPorts(self, urls):
+    """ Extract ports from urls
+
+        :param list urls: urls that can contain port, .e.g:: System/Service:port
+
+        :return: (dict, list)
+    """
+    portMapping = {}
+    newURLs = []
+    for _url in urls:
+      if ':' in _url:
+        urlTuple = _url.split(':')
+        if urlTuple[0] not in portMapping:
+          portMapping[urlTuple[0]] = urlTuple[1]
+        newURLs.append(urlTuple[0])
+      else:
+        newURLs.append(_url)
+    return (portMapping, newURLs)
```

Comment:
Could not you use `urllib.urlparse` in this method and in others? It would be less error prone than string manipulation.

Reply:
at this point, the URL format is as follows `System/Service:port`, `urlparse` will not work well here, as for me